nixos/modules/services/backup/burpClient.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, networking, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.burpClient;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.burpClient = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable burp.";
+      };
+
+      configFile = mkOption {
+        default = "/etc/burp/burp.conf";
```

Comment:
@tokudan if the program absolutely insists on keys in the configuration file, I believe your only option for achieving a declarative interface presently is to write a wrapper that generates the real configuration on the fly, substituting in passphrases/whatever read from files outside the store at run-time; `readFile` reads the named file into a string during evaluation, writing the resulting string to the store means the secret also ends up there, so it won't help you much beyond keeping the secret out of `configuration.nix` (which is an improvement in its own right, for sure, but short of what you're asking for).